### PR TITLE
add host_to_device() call on result of test_lower_dimensional_inputs

### PR DESCRIPTION
**Pull request by gronerl**
## Description

On my local machine, the test_lower_dimensional_inputs were failing due to missing copy from device

### CODE REVIEW

**Comment by gronerl** [comment]
bors try

**Comment by gronerl** [comment]
bors try

**Comment by gronerl** [comment]
The reason why this isn't caught by the CI is that in this test, GPU backends are not marked with requires_gpu, however the tests are excluded if cupy can't be imported (in which case the backends are not added to ALL_BACKENDS) so neither the gitlab nor the github pipelines run these tests.

**Comment by gronerl** [comment]
bors try

**Comment by gronerl** [comment]
bors retry

**Comment by gronerl** [comment]
bors try-
bors try

**Comment by gronerl** [comment]
bors try-

**Comment by gronerl** [comment]
bors try

**Comment by gronerl** [comment]
bors try

**Comment by gronerl** [comment]
bors try-
bors try

File: tests/definitions.py [hunk 1]
```diff
@@ -29,22 +29,61 @@
...
 
-ALL_BACKENDS = list(gt_backend.REGISTRY.keys())
-if cp is None:
-    # Skip gpu backends
-    ALL_BACKENDS = [
-        name for name in ALL_BACKENDS if gt_backend.from_name(name).storage_info["device"] != "gpu"
-    ]
+def _backend_name_as_param(name):
+    if gt_backend.from_name(name).storage_info["device"] == "gpu":
+        return pytest.param(name, marks=[pytest.mark.requires_gpu])
```
Comment (DropD) [inline tests/definitions.py:34]: Nice and elegant solution.

**Review by DropD — APPROVED** [review]

**Pull request merged**